doc/tutorial/using_multi_gpu.txt

```diff
@@ -0,0 +1,137 @@
+models between machines.
+
+.. warning::
+
+   The code is rather new and is still considered experimental at this
+   point.  It has been tested and seems to perform correctly in all
+   cases observed, but make sure to double-check your results before
+   publishing a paper or anything of the sort.
+
+
+Defining the context map
+------------------------
+
+The mapping from context names to devices is done through the
+:attr:`config.contexts` option.  The format looks like this::
+
+    dev0->cuda0;dev1->cuda1
+
+Let's break it down.  First there is a list of mappings.  Each of
+these mappings is separeted by a semicolon ';'.  There can be any
+number of such mappings, but in the example above we have two of them:
+`dev0->cuda0` and `dev1->cuda1`.
+
+The mappings themselves are composed of a context name followed by the
+two characters '->' and the device name.  The context name is a simple
+string which does not have any special meaning for Theano.  For
+parsing reasons, the context name cannot contain the sequence '->' or
+';'.  To avoid confusion context names that begin with 'cuda' or
+'opencl' as disallowed.  The device name is a device in the form that
+gpuarray expects like 'cuda0' or 'opencl0:0'.
+
+.. note::
+
+   Since there are a bunch of shell special characters in the syntax,
+   defining this on the command-line will require proper quoting, like this:
+
+   .. code-block:: shell
+
+       $ THEANO_FLAGS="contexts=dev0->cuda0"
+
+If you don't have enough GPUs for a certain model, you can assign the
+same device to more than one name. You can also assign extra names
+that a model doesn't need to some other devices.  However, a
+proliferation of names is not always a good idea since theano often
+assumes that different context names will be on different devices and
+will optimize accordingly.  So you may get faster performance for a
+single name and a single device.
+
+.. note::
+
+   It is often the case that multi-gpu operation requires or assumes
+   that all the GPUs involved are equivalent.  This is not the case
+   for this implementation.  Since the user has the task of
+   distrubuting the jobs across the different device a model can be
+   built on the assumption that one of the GPU is slower or has
+   smaller memory.
+
+
+A simple graph on two GPUs
+--------------------------
+
+The following simple program works on two GPUs.  It builds a function
+which perform two dot products on two different GPUs.
+
+.. code-block:: python
+
+   import numpy
+   import theano
+
+   v01 = theano.shared(numpy.random.random((1024, 1024)).astype('float32'),
+                       target='dev0')
+   v02 = theano.shared(numpy.random.random((1024, 1024)).astype('float32'),
+                       target='dev0')
+   v11 = theano.shared(numpy.random.random((1024, 1024)).astype('float32'),
+                       target='dev1')
+   v12 = theano.shared(numpy.random.random((1024, 1024)).astype('float32'),
```

Comment:
I needed to fix this in the file I created with this code. But my file currently have the same thing as this comment. Not sure what happened...